### PR TITLE
OZ-348: EIP Client made generic (OpenMRS-agnostic) + Upgrade to Java 17.

### PR DESCRIPTION
**Pull request by kipchubett**
Issue: https://mekomsolutions.atlassian.net/jira/software/c/projects/OZ/issues/OZ-348

In this PR, I have made several enhancements:
- Upgraded the project to Java 17, leveraging its new features and performance improvements.
- Refactored the EIP client to make it generic and remove the hard dependency on `openmrs-watcher`.
- Optimized the Docker image using Spring Boot layered tools.
- Enhanced the **README** documentation for better clarity and usage instructions.

### CODE REVIEW

**Review by mks-d — COMMENTED** [review]
So, by removing the `openmrs-watcher` dependency, this does not have to wait for [EIP-133](https://issues.openmrs.org/browse/EIP-133), correct?

**Comment by kipchubett** [comment]
> So, by removing the `openmrs-watcher` dependency, this does not have to wait for [EIP-133](https://issues.openmrs.org/browse/EIP-133), correct?

Yes.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ARG EIP_HOME_DIR=${EIP_CLIENT_DIR}/eip-home
+ARG EIP_ROUTES_DIR=${EIP_CLIENT_DIR}/routes
+
+RUN mkdir -p ${EIP_CONFIG_DIR} ${EIP_HOME_DIR} ${EIP_ROUTES_DIR}
+
+WORKDIR ${EIP_CLIENT_DIR}
+
+ENV EIP_PROFILE=dev
+
+RUN chmod +x ./start.sh && chmod +x ./version.txt
```
Comment (ibacher) [inline Dockerfile]:
```suggestion
RUN chmod +x ./start.sh
```

File: app/src/main/resources/assembly/assembly.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        <format>tar</format>
+        <format>zip</format>
+    </formats>
+    <fileSets>
+        <fileSet>
+            <directory>src/main/resources/scripts</directory>
+            <outputDirectory>.</outputDirectory>
+            <includes>
+                <include>start.sh</include>
+                <include>start.bat</include>
```
Comment (ibacher) [inline app/src/main/resources/assembly/assembly.xml]: General remark: I think we have both `start.bat` and `start.cmd` for the Windows file in different repos. We should stick to one (I have no preference).

**Review by ibacher — APPROVED** [review]
Basically LGTM.

File: app/src/main/resources/assembly/assembly.xml (hunk 2 above)
Comment (kipchubett) [reply app/src/main/resources/assembly/assembly.xml]: Then we should stick with `.cmd`, they are modern batch files(using in Windows NT and later versions) which offers more capabilities, additional commands, better scripting support, and enhanced error handling compared to `.bat` files.

**Review by Ruhanga — APPROVED** [review]
LGTM.

**Pull request merged**